src/tokens/slice.ts

```diff
@@ -129,6 +138,41 @@ const slice = createSlice({
       loading: false,
       error: false,
     }),
+    fetchPriceHistoryStart: (
+      state: Draft<State>,
+      action: PayloadAction<{
+        tokenId: string
+        startTimestamp: number
+        endTimestamp: number
+      }>
+    ) => {
+      const { tokenId } = action.payload
+      const token = state.tokenBalances[tokenId]
+      if (token) {
+        token.historicalPricesUsd = {
+          ...token.historicalPricesUsd,
+          priceHistoryStatus: 'loading',
+        }
+      }
+    },
+    fetchPriceHistorySuccess: (
+      state: Draft<State>,
+      action: PayloadAction<{ tokenId: string; priceHistory: TokenPriceHistoryEntry[] }>
+    ) => {
+      const { tokenId, priceHistory } = action.payload
+      const token = state.tokenBalances[tokenId]
+      if (priceHistory && token?.historicalPricesUsd) {
```

Comment:
and if there are no existing historical prices in state, the new ones get ignored? that doesn't seem right

Reply:
At this point if there are no existing historical prices in the state it'll still have the priceHistoryStatus in historicalPricesUsd so it'll load prices.
```TypeScript
'celo-mainnet:native': {
  ...StoredTokenBalance
  historicalPricesUsd: {
    priceHistoryStatus: 'loading',
  }
}
```